lib/rhai/engine.ex

```diff
@@ -47,6 +47,28 @@ defmodule Rhai.Engine do
     Rhai.Native.engine_eval(resource, script)
   end
 
+  @doc """
+  Set the maximum length of arrays (0 for unlimited).
+
+  Not available under `unchecked` or `no_index`.
+  """
+  @spec set_fail_on_invalid_map_property(t(), integer()) :: t()
+  def set_max_array_size(%__MODULE__{resource: resource} = engine, max_size) do
+    Rhai.Native.engine_set_max_array_size(resource, max_size)
+
+    engine
+  end
+
+  @doc """
+  The maximum length of arrays (0 for unlimited).
+
+  Zero under `no_index`.
+  """
+  @spec fail_on_invalid_map_property?(t()) :: integer()
```

Comment:
remove `?` since it;'s returning an integer. You could also use `non_neg_integer()` as a type here.